packages/config/src/ProfileInfo.ts

```diff
@@ -464,10 +483,16 @@ export class ProfileInfo {
             }
 
             if (missingRequired.length > 0) {
-                throw new ImperativeError({ msg: "Missing required properties: " + missingRequired.join(", ") });
+                throw new ProfInfoErr({
+                    errorCode: ProfInfoErr.MISSING_REQ_PROP,
+                    msg: "Missing required properties: " + missingRequired.join(", ")
+                });
```

Comment:
Perhaps this validation is unnecessary and we should leave it up to SDK users to throw an exception if options they deem to be required are missing?

This question isn't directly related to the PR but I wanted to raise it for discussion. If we think this validation is unnecessary, removing it doesn't need to be addressed in this PR - I can create an issue for it 🙂

Reply:
Since the the array of missingArgs contains arguments with default values, perhaps we should check for both missingArgs.length > 0 and at least one missingArg has no value?

If default values were added to knownArgs, the consuming app could make an easy test for missingArgs.length > 0 and would not need our exception.

If we can concisely describe what we could give to the consumer, maybe we should ask the consumers what they would prefer in our design doc Github issue?

Reply:
> If we can concisely describe what we could give to the consumer, maybe we should ask the consumers what they would prefer in our design doc Github issue?

Agree this might be a good idea. For context, the current implementation is that we check if the number of missingArgs in the profile schema's `required` property is greater than 0.